src/transaction/id.rs

```diff
@@ -58,39 +73,42 @@ impl Transaction {
             }
         });
 
-        self.outputs_mut()
-            .iter_mut()
-            .for_each(|output| match output {
-                Output::Contract {
-                    balance_root,
-                    state_root,
-                    ..
-                } => {
-                    balance_root.iter_mut().for_each(|b| *b = 0);
-                    state_root.iter_mut().for_each(|b| *b = 0);
-                }
+        outputs.iter_mut().for_each(|output| match output {
+            Output::Contract {
+                balance_root,
+                state_root,
+                ..
+            } => {
+                balance_root.iter_mut().for_each(|b| *b = 0);
+                state_root.iter_mut().for_each(|b| *b = 0);
+            }
 
-                Output::Change { amount, .. } => *amount = 0,
+            Output::Change { amount, .. } => *amount = 0,
 
-                Output::Variable {
-                    to, amount, color, ..
-                } => {
-                    to.iter_mut().for_each(|b| *b = 0);
-                    *amount = 0;
-                    color.iter_mut().for_each(|b| *b = 0);
-                }
+            Output::Variable {
+                to, amount, color, ..
+            } => {
+                to.iter_mut().for_each(|b| *b = 0);
+                *amount = 0;
+                color.iter_mut().for_each(|b| *b = 0);
+            }
 
-                _ => (),
-            });
+            _ => (),
+        });
+
+        witnesses.clear();
```

Comment:
This is already done on line 45: `self.set_witnesses(vec![]);`

Reply:
True, added this commit to not waste the allocated capacity of the vector:

https://github.com/FuelLabs/fuel-tx/pull/57/commits/b49ea0221c8f3d681dad286e2c8d71eb31f9190f